sdk/scheduler/src/main/java/com/mesosphere/sdk/http/endpoints/HealthResource.java

```diff
@@ -1,79 +1,582 @@
 package com.mesosphere.sdk.http.endpoints;
 
 import com.mesosphere.sdk.http.ResponseUtils;
-import com.mesosphere.sdk.scheduler.SchedulerConfig;
+import com.mesosphere.sdk.scheduler.plan.Plan;
 import com.mesosphere.sdk.scheduler.plan.PlanCoordinator;
 import com.mesosphere.sdk.scheduler.plan.PlanManager;
+import com.mesosphere.sdk.state.FrameworkStore;
+import com.mesosphere.sdk.state.StateStoreException;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.mesos.Protos;
+import org.json.JSONArray;
+import org.json.JSONObject;
 
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.Response;
 
-import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
 import java.util.stream.Collectors;
 
+// CHECKSTYLE:OFF LineLengthCheck
+
 /**
- * A read-only API for checking whether the service is healthy.
+ * This class returns a single code representing the status of the service
```

Comment:
I am inclined to say to delete this javadoc, The `ServiceStatusCode` enum would serve as single source of truth.